### PR TITLE
make govet linter pass

### PR DESCRIPTION
**Pull request by melekes**
Refs #3262

### CODE REVIEW

File: crypto/secp256k1/secp256k1_nocgo.go [hunk 1]
```diff
@@ -52,8 +52,8 @@ func (pubKey PubKeySecp256k1) VerifyBytes(msg []byte, sigStr []byte) bool {
 // that len(sigStr) == 64.
 func signatureFromBytes(sigStr []byte) *secp256k1.Signature {
 	return &secp256k1.Signature{
-		new(big.Int).SetBytes(sigStr[:32]),
-		new(big.Int).SetBytes(sigStr[32:64]),
+		R: new(big.Int).SetBytes(sigStr[:32]),
```
Comment (melekes) [inline crypto/secp256k1/secp256k1_nocgo.go]: @Liamsi is this correct?

File: p2p/conn/secret_connection_test.go [hunk 2]
```diff
@@ -245,11 +245,11 @@ func TestSecretConnectionReadWrite(t *testing.T) {
 						}
 						*nodeReads = append(*nodeReads, string(readBuffer[:n]))
 					}
-					if err := nodeConn.PipeReader.Close(); err != nil {
-						t.Error(err)
-						return nil, err, true
-					}
-					return nil, nil, false
+					// if err := nodeConn.PipeReader.Close(); err != nil {
```
Comment (melekes) [inline p2p/conn/secret_connection_test.go]: remove?

File: crypto/encoding/amino/encode_test.go [hunk 3]
```diff
@@ -47,6 +47,8 @@ func checkAminoJSON(t *testing.T, src interface{}, dst interface{}, isNil bool)
 	require.Nil(t, err, "%+v", err)
 }
 
+// ExamplePrintRegisteredTypes refers to unknown identifier: PrintRegisteredTypes
+//nolint:govet
 func ExamplePrintRegisteredTypes() {
```
Comment (melekes) [inline crypto/encoding/amino/encode_test.go]: can't find PrintRegisteredTypes

File: libs/common/bit_array.go [hunk 4]
```diff
@@ -412,6 +412,7 @@ func (bA *BitArray) UnmarshalJSON(bz []byte) error {
 			bA2.SetIndex(i, true)
 		}
 	}
-	*bA = *bA2
+	// assignment copies lock value to *bA: github.com/tendermint/tendermint/libs/common.BitArray contains sync.Mutex
+	*bA = *bA2 //nolint:govet
```
Comment (melekes) [inline libs/common/bit_array.go]: ok to copy the lock?

File: crypto/secp256k1/secp256k1_nocgo.go (hunk 1 above)
Comment (Liamsi) [reply crypto/secp256k1/secp256k1_nocgo.go]: Yes, that is correct!

File: crypto/encoding/amino/encode_test.go (hunk 3 above)
Comment (Liamsi) [reply crypto/encoding/amino/encode_test.go]: https://github.com/golang/go/issues/29570
(fixed on tip)

File: p2p/conn/secret_connection_test.go (hunk 2 above)
Comment (Liamsi) [reply p2p/conn/secret_connection_test.go]: You can definitely remove without changing the existing semantics here. What the developer probably wanted, was that the connection gets closed on an error/when the control flow exits the loop. Sounds like this clsoing code should be moved to a defer (using the outer function though)?

Comment (Liamsi) [reply p2p/conn/secret_connection_test.go]:
```go
defer func() {
				if err := nodeConn.PipeReader.Close(); err != nil {
					t.Error(err)
				}
			}()
```
on line 206?

File: libs/common/bit_array.go (hunk 4 above)
Comment (Liamsi) [reply libs/common/bit_array.go]: I think  it is OK here: the object is just being constructed ... If you want to remove the `nolint`, you could get rid of `bA2` here instead and init the fields of `bA` directly:
```go
        bA.Bits = numBits
	bA.Elems = make([]uint64, (numBits+63)/64)
	for i := 0; i < numBits; i++ {
		if bits[i] == 'x' {
			bA.SetIndex(i, true)
		}
	}
```

Comment (melekes) [reply libs/common/bit_array.go]: no, I think it's fine to leave nolint

File: p2p/conn/secret_connection_test.go (hunk 2 above)
Comment (melekes) [reply p2p/conn/secret_connection_test.go]: https://github.com/tendermint/tendermint/pull/3292/commits/b38cbae53df73ac9b712d56431968931baf14e7c

**Pull request merged**